crates/rolldown/src/ast_scanner/side_effect_detector/annotation.rs

```diff
@@ -1,10 +1,14 @@
+use daachorse::DoubleArrayAhoCorasick;
 use once_cell::sync::Lazy;
 use oxc::span::Span;
 
 use super::SideEffectDetector;
 
-static PURE_COMMENTS: Lazy<regex::Regex> =
-  Lazy::new(|| regex::Regex::new("^\\s*(#|@)__PURE__\\s*$").expect("Should create the regex"));
+static PURE_COMMENTS: Lazy<DoubleArrayAhoCorasick<usize>> = Lazy::new(|| {
```

Comment:
according to https://rollupjs.org/configuration-options/#pure:~:text=Comments%20containing%20%40__PURE__%20or%20%23__PURE__%20mark%20a%20specific%20function%20call%20or, it is contain `__PURE__` not only pure and whitespace.

Reply:
In this scenario, using `DoubleArrayAhoCorasick` should search faster, use less memory, and have shorter initiative time.